sdk/identity/identity/src/credentials/managedIdentityCredential/utils.ts

```diff
@@ -7,10 +7,14 @@ import { IdentityClient } from "../../client/identityClient";
 import { DefaultScopeSuffix } from "./constants";
 import { MSIExpiresInParser } from "./models";
 
+/**
+ * Transforms the received scope or array of scopes from getToken to a resource string.
+ * We only keep the first scope we receive, and we remove the permissions or the /.default path after the domain.
```

Comment:
This can backfire though: a caller who passes two scopes gets a token for the first scope and may be surprised by the authentication error following a client call requiring a token for the second scope. In other SDKs, ManagedIdentityCredential throws an exception when asked to get a token for multiple scopes. (The token exchange credential shouldn't do this, however, because it doesn't have the same one-scope limitation.)

Reply:
Ohh so this is why we were throwing an error before. Thank you for letting me know! It seemed unnecessarily strict. I’ll add your note on the comments.

Reply:
I need a moment to clean this up…

Reply:
Updated!